sled-agent/src/services.rs

```diff
@@ -212,16 +206,31 @@ impl Config {
             sidecar_revision,
             gateway_address,
             all_svcs_config_path: default_services_config_path(),
-            get_svc_config_dir: Box::new(|zone_name: &str, svc_name: &str| {
-                PathBuf::from(ZONE_ZFS_DATASET_MOUNTPOINT)
-                    .join(PathBuf::from(zone_name))
-                    .join("root")
-                    .join(format!("var/svc/manifest/site/{}", svc_name))
-            }),
         }
     }
 }
 
+// A wrapper around `ZoneRequest`, which allows it to be serialized
```

Comment:
Is this configuration that is only used during development? Is it something that goes away with self-assembling zones or is this actually part of making self-assembling zones work?

Reply:
This is unfortunately not a development-only piece of metadata.

Here's my primer on "old vs new" zone management:

## Old

- All zone filesystems are provisioned in `rpool/zone` (the ramdisk)
- Zones are **not** self-assembling, meaning that they need tweaking from the sled agent after booting
- The list of zones under the purview of the sled are stored in a config file in `/var/oxide` (... unfortunately, this is also in the ramdisk)

## New (not totally done in this PR, but we're heading this direction)

- Zone filesystems are provisioned in `/pool/ext/<UUID>/zone` on particular U.2s
- Zones are self-assembling, meaning that once they are created, they can be booted
  - NOTE: The sled agent may still need to supply some metadata prior to running the zone, such as creating VNICs!
- The list of zones under the purview of the sled agent are stored in `/pool/int/<UUID>/config` on both M.2s


## So this metadata...

Yeah, so this piece of metadata is here because we're breaking the previous standard that "every zone filesystem lives in `/zone`". By shuffling that around, the sled agent actually needs to know where to find zones to launch them.

I admittedly **could** store this information alongside the zone filesystem on the U.2, but by storing it within the `config` directory of the M.2, I figured we'd have a better ability for sleds to eventually notice / report "hey, I **should** launch a particular zone, but it hasn't come up yet because that disk is gone".